### PR TITLE
Improve Test Coverage for ToClaudeRequest Function from OpenAI-Claude Converter

### LINKED ISSUE #85: Add tests for the package
Tests should be added to the package, to cover all functionalities.

### PR DESCRIPTION
**Pull request by nvtrinh2001**
## Background   

- Issue: #85  
- This PR enhances the test suite for the `ToClaudeRequest` function to ensure robust request conversion and error handling.  

## Changes   

- **Added tests for valid request conversions:**  
  - Standard request conversion.  
  - Default `max_tokens` handling based on model.  
  - Handling of system messages.  
  - Handling of `TopP` parameter.  
  - Support for tools and tool choice.  

- **Introduced error handling tests:**  
  - Missing messages.  
  - Empty request.  
  - Unsupported response format.  

- **Validated special cases:**  
  - Handling of stop sequences.  

## Considerations   

- This branch depends on PR #86, so please review it first.  
- Ensure test coverage includes all possible edge cases to prevent regressions.  

## Remaining Tasks

### CODE REVIEW

File: converter/openaiclaudeconverter/openai_to_claude_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		assert.NotNil(t, result.ToolChoice)
+	})
+
+	t.Run("fails with missing messages", func(t *testing.T) {
+		openaiRequest := &openai.ChatCompletionRequest{
+			Model: "claude-3-sonnet",
+		}
+
+		result, err := ToClaudeRequest(openaiRequest)
+		assert.Error(t, err)
```
Comment (yoonkon-nam) [inline converter/openaiclaudeconverter/openai_to_claude_test.go:126]: Could you please confirm the error message? not just check error exists?

Comment (nvtrinh2001) [reply converter/openaiclaudeconverter/openai_to_claude_test.go]: Updated with error checking.
